### PR TITLE
SONARIAC-973 ARM Bicep add ifCondition to resourceDecl

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java [hunk 1]
```diff
@@ -188,7 +188,9 @@ public ResourceDeclaration RESOURCE_DECLARATION() {
         INTERPOLATED_STRING(),
         b.optional(b.token(BicepKeyword.EXISTING)),
         b.token(Punctuator.EQU),
-        OBJECT_EXPRESSION(),
+        b.firstOf(
+          OBJECT_EXPRESSION(),
+          IF_EXPRESSION()),
```
Comment (petertrr) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java:193]: We also already have `FOR_EXPRESSION`, should it be supported in here in a separate PR or in this one as well?

**Review by petertrr — APPROVED** [review]
LGTM!

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java:193]: That's a fair point, I actually tried to implement both `IF_EXPRESSION` and `FOR_EXPRESSION`, but I couldn't make it work properly as there was something missing on `FOR_EXPRESSION` part, it would have required an additional modification later.
So in the end I didn't want to partially do it for `ForExpression` and just kept separate ticket :smile:
